processing/src/main/java/io/druid/query/groupby/GroupByQueryEngine.java

```diff
@@ -312,12 +314,113 @@ public RowIterator(GroupByQuery query, Cursor cursor, ByteBuffer metricsBuffer,
       dimensionSpecs = query.getDimensions();
       dimensions = Lists.newArrayListWithExpectedSize(dimensionSpecs.size());
       dimNames = Lists.newArrayListWithExpectedSize(dimensionSpecs.size());
+
       for (int i = 0; i < dimensionSpecs.size(); ++i) {
         final DimensionSpec dimSpec = dimensionSpecs.get(i);
-        final DimensionSelector selector = cursor.makeDimensionSelector(dimSpec.getDimension());
-        if (selector != null) {
-          dimensions.add(selector);
+        final String dimension = dimSpec.getDimension();
+        if(dimension.equals(Column.TIME_COLUMN_NAME)) {
```

Comment:
I'm wondering if this logic shouldn't be given to the cursor and done inside of `makeDimensionSelector`.  I'm assuming that you chose not to do it that way because that means each implementation of `StorageAdapter` has to implement the time lookup properly, but it also seems like each implementation of `StorageAdapter` might be able to take advantage of optimizations when returning the time column...

I guess it seems like we are building in extra logic about "the time column acts like XYZ", where I think that with 0.7 there was an effort to make the time column just look like any other column.  This seems to be breaking that "similarity"...